### PR TITLE
OCM-6574 | feat: Added trusted ip list to tf

### PR DESCRIPTION
**Pull request by den-rgb**
**What this PR does / why we need it**:

Request from [OCM-6084](https://issues.redhat.com/browse/OCM-6084) , Feature adding ability to output list of Red Hat trusted ip addresses + tests + documentation and examples.

JIRA: [OCM-6574](https://issues.redhat.com/browse/OCM-6574)

**Change type**
- [x] New feature
- [ ] Bug fix
- [ ] Build
- [ ] CI
- [x] Documentation
- [ ] Performance
- [ ] Refactor
- [ ] Style
- [ ] Unit tests
- [x] Subsystem tests

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.

### CODE REVIEW

**Comment by gdbranco** [comment]
Could you add a test to it?

File: provider/trusted_ip_addresses/trusted_ip_addresses_data_source.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+		Description: "List of trusted ip addresses",
+		Attributes: map[string]schema.Attribute{
+			"page": schema.Int64Attribute{
+				Description: "Page number of the result set.",
+				Computed:    true,
+			},
+			"size": schema.Int64Attribute{
+				Description: "Number of items per page.",
+				Computed:    true,
+			},
```
Comment (nirarg) [inline provider/trusted_ip_addresses/trusted_ip_addresses_data_source.go:55]: Its not clear to me why the pagination info should be exposed to the user.
This is internal implementation info and the user shouldn't have this in the output.
What do you think?

File: provider/trusted_ip_addresses/trusted_ip_addresses_data_source.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+			"total": schema.Int64Attribute{
+				Description: "Total number of items in the result set.",
+				Computed:    true,
+			},
+			"items": schema.ListNestedAttribute{
+				Description: "Items of the list.",
```
Comment (nirarg) [inline provider/trusted_ip_addresses/trusted_ip_addresses_data_source.go:61]:
```suggestion
				Description: "List of all trusted IP addresses.",
```

File: provider/trusted_ip_addresses/trusted_ip_addresses_data_source.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+	}
+
+	for _, listItem := range listItems {
+		trustedIpState := &TrustedIpState{
+			ID:      basetypes.NewStringValue(listItem.ID()),
+			Enabled: basetypes.NewBoolValue(listItem.Enabled()),
+		}
+
+		state.Items = append(state.Items, trustedIpState)
+	}
```
Comment (nirarg) [inline provider/trusted_ip_addresses/trusted_ip_addresses_data_source.go:147]: This is a convert action between OCM object to tf object
Maybe it more appropriate to move this to its own function
Something like:
`trustedIpState := fromOCMListItems(listItems)`

File: provider/trusted_ip_addresses/trusted_ip_addresses_data_source.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+				Computed:    true,
+			},
+			"items": schema.ListNestedAttribute{
+				Description: "Items of the list.",
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"id": schema.StringAttribute{
+							Description: "Ip address.",
+							Computed:    true,
+						},
```
[the remark this answers is not in the document]
Comment (den-rgb) [reply provider/trusted_ip_addresses/trusted_ip_addresses_data_source.go:67]: On cluster service it is stored as `id`, I'm just trying to keep it consistent `ocm get /api/clusters_mgmt/v1/trusted_ip_addresses`

File: provider/trusted_ip_addresses/trusted_ip_addresses_data_source.go (hunk 1 above)
Comment (den-rgb) [reply provider/trusted_ip_addresses/trusted_ip_addresses_data_source.go:55]: I'll update to just include the total and Items

File: docs/data-sources/trusted_ip_addresses.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "rhcs_trusted_ip_addresses Data Source - terraform-provider-rhcs"
+subcategory: ""
+description: |-
+  List of trusted ip addresses
+---
+
+# rhcs_trusted_ip_addresses (Data Source)
+
+List of trusted ip addresses
```
[the remark this answers is not in the document]
Comment (den-rgb) [reply docs/data-sources/trusted_ip_addresses.md:11]: Example usage doesn't seem to generate despite adding it to the generate_example_usages/data_sources dir

[the remark this answers is not in the document]
Comment (den-rgb) [reply docs/data-sources/trusted_ip_addresses.md:11]: All there really is is this blog: https://source.redhat.com/personal_blogs/listing_red_hats_trusted_ip_addresses_using_an_api_enpoint

File: examples/list_trusted_ip_addresses/main.tf [hunk 6]
```diff
@@ -0,0 +1,35 @@
+#
```
[the remark this answers is not in the document]
Comment (den-rgb) [reply examples/list_trusted_ip_addresses/main.tf:1]: https://github.com/terraform-redhat/terraform-rhcs-rosa-classic/pull/68

[the remark this answers is not in the document]
Comment (den-rgb) [reply examples/list_trusted_ip_addresses/main.tf:1]: PTAL

**Pull request merged**